account_parent/models/account.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2016 Steigend IT Solutions
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import odoo.addons.decimal_precision as dp
+from odoo import api, fields, models
+
+
+class AccountAccountType(models.Model):
+    _inherit = "account.account.type"
+
+    type = fields.Selection(selection_add=[('view', 'View')])
+
+
+class AccountAccount(models.Model):
+    _inherit = "account.account"
+
+    @api.model
+    def _move_domain_get(self):
+        context = dict(self._context or {})
+
+        res = ''
+        date_field = 'date'
+        if context.get('aged_balance'):
+            date_field = 'date_maturity'
+        if context.get('date_to'):
+            res += " AND l.{} <= '{}' ".format(date_field, context['date_to'])
+        if context.get('date_from'):
+            if not context.get('strict_range'):
+                res += " AND ({} >= '{}' ".format(
+                    date_field, context['date_from'], )
+                res += " OR at.include_initial_balance = true) "
+            elif context.get('initial_bal'):
+                res += " AND l.{} < '{}' ".format(
+                    date_field, context['date_from'])
+            else:
+                res += " AND l.{} >= '{}' ".format(
+                    date_field, context['date_from'])
+
+        if context.get('journal_ids'):
+            res += " AND l.journal_id in {} ".format(
+                tuple(context['journal_ids']))
+
+        state = context.get('state')
+        if state and state.lower() != 'all':
+            res += " AND m.state = '{}' ".format(state)
+
+        if context.get('company_id'):
+            res += " AND l.company_id = {} ".format(context['company_id'])
+
+        if 'company_ids' in context:
+            res += " AND l.company_id  in {} ".format(
+                tuple(context['company_ids']))
+
+        return res
+
+    @api.multi
+    @api.depends('move_line_ids', 'move_line_ids.amount_currency',
+                 'move_line_ids.debit', 'move_line_ids.credit')
+    def _compute_values(self):
+        default_domain = self._move_domain_get()
+        query = 'SELECT sum(debit) debit, sum(credit) credit, ' \
+                'sum(debit - credit) balance ' \
+                'FROM account_move_line l ' \
+                'LEFT JOIN account_move m ON l.move_id = m.id ' \
+                'LEFT JOIN account_account a ON l.account_id = a.id ' \
+                'LEFT JOIN account_account_type at ON a.user_type_id = at.id '\
+                'WHERE 1 = 1 AND l.account_id in %s '
+        query += default_domain
+        for account in self:
+            sub_accounts = self.with_context(
+                {'show_parent_account': True}).search([
+                    ('id', 'child_of', [account.id])])
+            params = [tuple(sub_accounts.ids)]
+            self.env.cr.execute(query, params)
```

Comment:
That's funtion fields will be automated compute value. I think it don't the same when you click on button and compute value of all account.

The wizard no call to that function. It just show hierarchy view with already exist data. When you find any account you also can see debit, credit, balance.